toolkit/scripts/toolkit.mk

```diff
@@ -12,36 +12,90 @@ toolkit_component_extra_files = \
+
+$(call create_folder,"$(rpms_snapshot_build_dir)")
+$(call create_folder,"$(toolkit_prep_dir)")
 
 # Outputs
-toolkit_version   = $(RELEASE_VERSION)-$(build_arch)
-toolkit_archive   = $(OUT_DIR)/toolkit-$(toolkit_version).tar.gz
-toolkit_remove_archive = $(OUT_DIR)/toolkit-*.tar.gz
-toolkit_build_dir = $(BUILD_DIR)/toolkit
-toolkit_repos_dir = $(toolkit_build_dir)/repos
-toolkit_tools_dir = $(toolkit_build_dir)/tools/toolkit_bins
-toolkit_release_file = $(toolkit_build_dir)/version.txt
+toolkit_archive_versioned_compressed   = $(OUT_DIR)/toolkit-$(toolkit_version).tar.gz
+rpms_snapshot = $(OUT_DIR)/$(rpms_snapshot_name)
 
-.PHONY: package-toolkit clean-package-toolkit
+.PHONY: package-toolkit rpms-snapshot clean-package-toolkit clean-rpms-snapshot
+
+clean: clean-package-toolkit clean-rpms-snapshot
 
-clean: clean-package-toolkit
 clean-package-toolkit:
 	rm -f $(toolkit_remove_archive)
 	rm -rf $(toolkit_build_dir)
 
-package-toolkit: go-tools
-	rm -rf $(toolkit_build_dir) && \
-	mkdir -p $(toolkit_build_dir) && \
+clean-rpms-snapshot:
+	rm -f $(rpms_snapshot)
+	rm -rf $(rpms_snapshot_build_dir)
+	rm -r $(rpms_snapshot_logs_path)
+
+package-toolkit: $(toolkit_archive_versioned_compressed)
+	@echo "Toolkit packed under '$(toolkit_archive_versioned_compressed)'."
+
+$(toolkit_archive_versioned_compressed): $(toolkit_archive) $(rpms_snapshot) $(depend_SPECS_DIR)
+	cp $(toolkit_archive) $(toolkit_archive_versioned) && \
+	echo "$(toolkit_version)" > $(toolkit_release_file) && \
+	cp $(rpms_snapshot) $(toolkit_rpms_snapshot_file) && \
+	tar --update -f $(toolkit_archive_versioned) -C $(toolkit_build_dir) $(toolkit_release_file_relative_path) $(toolkit_rpms_snapshot_file_relative_path) && \
+	$(ARCHIVE_TOOL) --best -c $(toolkit_archive_versioned) > $(toolkit_archive_versioned_compressed)
+
+$(toolkit_archive): $(go_tool_targets) $(mariner_repos_files) $(toolkit_component_extra_files) $(toolkit_root_files)
+	rm -rf $(toolkit_prep_dir) && \
+	mkdir -p $(toolkit_prep_dir) && \
 	mkdir -p $(toolkit_repos_dir) && \
 	mkdir -p $(toolkit_tools_dir) && \
-	cp -r $(toolkit_root)/* $(toolkit_build_dir) && \
-	cp $(mariner_repos_dir)/*.repo $(toolkit_repos_dir) && \
-	cp $(toolkit_component_extra_files) $(toolkit_build_dir) && \
+	cp -r $(toolkit_root_files) $(toolkit_prep_dir) && \
+	cp $(mariner_repos_files) $(toolkit_repos_dir) && \
+	cp $(toolkit_component_extra_files) $(toolkit_prep_dir) && \
 	cp $(go_tool_targets) $(toolkit_tools_dir) && \
-	echo "$(toolkit_version)" > $(toolkit_release_file) && \
-	rm -rf $(toolkit_build_dir)/out && \
-	tar -I $(ARCHIVE_TOOL) -cvp -f $(toolkit_archive) -C $(toolkit_build_dir)/.. $(notdir $(toolkit_build_dir))
+	rm -rf $(toolkit_prep_dir)/out && \
+	tar -cvp -f $(toolkit_archive) -C $(dir $(toolkit_prep_dir)) $(notdir $(toolkit_prep_dir))
+
+rpms-snapshot: $(rpms_snapshot)
+	@echo "RPMs snapshot generated under '$(rpms_snapshot)'."
+
+$(rpms_snapshot): $(rpms_snapshot_per_specs) $(depend_SPECS_DIR)
```

Comment:
I think this `$(rpms_snapshot):` middle step can be omitted, we could just have `$(rpms_snapshot_per_specs)` be `$(OUT_DIR)/$(rpms_snapshot_name)` from the get-go, right?

Reply:
I wanted to achieve two things:

1. Rebuilding existing snapshots on `SPECS_DIR` changes and
2. still have only one snapshot file for the current `SPECS_DIR` inside the output folder.
 
It's a minor thing, but to keep things clean, I wanted to make sure that you can always rely on one set output file to contain your results after running the `rpms-snapshot` target. It makes no difference inside out toolkit, since you can always have other targets ask for `rpms_snapshot_per_specs`, but any external dependence on the snapshot's location would have a harder time finding it.